ludwig/visualize.py

```diff
@@ -1119,8 +1526,8 @@ def compare_classifiers_performance_from_prob(
 
 
 def compare_classifiers_performance_from_pred(
-        predictions_per_model: List[list],
-        ground_truth: np.array,
+        predictions_per_model: List[np.ndarray],
+        ground_truth: np.ndarray,
```

Comment:
this is true everywhere:
both predictions and ground truth should be of the types that allow an API interface like this:
(pseudocode)
```
trainset, valset, testset, ... = moedl.train(dataset)
preds = model.predict(testset)
viz_function(ground_truth=testset[output_feature_name], predictions=[preds[output_feature_name][PREDICTIONS]])
```
or something that looks like this. So I'm not sure if ndarray is the correct one.

Reply:
@w4nderlust   I want to confirm my understanding of the guidance.  Following is a mock up of how I would move code from _cli function to the related viz function.  Does this look right?

#### Currently a _cli and its related viz function looks like this
```
def compare_classifiers_performance_from_prob_cli(
        probabilities: Union[str, List[str]],
        ground_truth: str,
        ground_truth_split: int,
        split_file: str,
        ground_truth_metadata: str,
        output_feature_name: str,
        output_directory: str,
        **kwargs: dict
) -> None:

    # retrieve feature metadata to convert raw predictions to encoded value
    metadata = load_json(ground_truth_metadata)

    # translate string to encoded numeric value
    # retrieve ground truth from source data set
    ground_truth = _extract_ground_truth_values(
        ground_truth,
        output_feature_name,
        ground_truth_split,
        split_file
    )

    feature_metadata = metadata[output_feature_name]
    vfunc = np.vectorize(_encode_categorical_feature)
    ground_truth = vfunc(ground_truth, feature_metadata['str2idx'])

    probabilities_per_model = load_data_for_viz(
        'load_from_file', probabilities, dtype=float
    )

    compare_classifiers_performance_from_prob(
        probabilities_per_model,
        ground_truth,
        output_directory=output_directory,
        **kwargs
    )


def  compare_classifiers_performance_from_prob():
   #
   # code to create the visualization
   #
```

#### My interpretation of the guidance "both predictions and ground truth should be of the types that allow an API interface like this:" changes to the following:
```
def compare_classifiers_performance_from_prob_cli(
        probabilities: Union[str, List[str]],
        ground_truth: str,
        ground_truth_split: int,
        split_file: str,
        ground_truth_metadata: str,
        output_feature_name: str,
        output_directory: str,
        **kwargs: dict
) -> None:

    compare_classifiers_performance_from_prob(
        probabilities,
        ground_truth,
        ground_truth_split,
        split_file,
        ground_truth_metadata,
        output_feature_name,
        output_directory=output_directory,
        **kwargs
    )


def  compare_classifiers_performance_from_prob(
    probabilities: str,  # file path string
    ground_truth: pd.Series, # pandas Series for the ground truth
    ground_truth_split: int,
    split_file: str,  #file path to split file
    ground_truth_metadata: str,  #file path to training_meta_data.json 
    output_directory: str,  #directory to store the visualization
):

   # following functions moved from the _cli to the viz function
    # retrieve feature metadata to convert raw predictions to encoded value
    metadata = load_json(ground_truth_metadata)

    # translate string to encoded numeric value
    # retrieve ground truth from source data set
    ground_truth = _extract_ground_truth_values(
        ground_truth,
        output_feature_name,
        ground_truth_split,
        split_file
    )

    feature_metadata = metadata[output_feature_name]
    vfunc = np.vectorize(_encode_categorical_feature)
    ground_truth = vfunc(ground_truth, feature_metadata['str2idx'])

    probabilities_per_model = load_data_for_viz(
        'load_from_file', probabilities, dtype=float
    )
   # end of functions moved from _cli to viz function

   #
   # code to create the visualization
   #

```

Reply:
With the last pushes, this should be taken care of.